lib/provider/aws/lambda-edge-origin-request/sanitize-headers.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const setCookieVariations = require('./set-cookie.json').variations;
+const readOnlyHeaders = [
+  "memoept-encoding",
+  "content-length",
+  "if-modified-since",
+  "if-none-Match",
+  "if-range",
+  "if-unmodified-since",
+  "range",
+  "transfer-encoding",
+  "via"
+];
+
+module.exports = function sanitizeHeaders(headers) {
+  return Object.keys(headers).reduce((memo, key) => {
+    const value = headers[key];
+    const normalizedKey = key.toLowerCase();
+
+    if (readOnlyHeaders.includes(normalizedKey)) {
+        return memo;
+    }
+    
+    if (memo[normalizedKey]) {
+      memo[normalizedKey].push({
+          key: key,
+          value: value
+      });
+
+      return memo;
+    }
+
+    if (Array.isArray(value) && normalizedKey === 'set-cookie') {
+      value.forEach((cookie, i) => {
+        memo[setCookieVariations[i]] = cookie;
```

Comment:
This seems like a bug: if you have an array of 2 cookies, this results in the following headers being set:

```json
"connection": [
    {
        "key": "connection",
        "value": "close"
    }
],
"set-cookie": "cookie1=hello",
"Set-cookie": "cookie2=world",
"foo": [
    {
        "key": "foo",
        "value": "bar"
    }
],
```

... because you're taking the index of the cookie, and assigning it to the Nth cookie-variant name... so `0` is `set-cookie`, `1` is `Set-cookie`, `2` is `sEt-cookie`, and so forth...

---

I would recommend:

1. Delete `set-cookie.json`.
2. Change this whole file (`sanitize-headers.js`) to:

```javascript
'use strict';

const readOnlyHeaders = [
  "memoept-encoding",
  "content-length",
  "if-modified-since",
  "if-none-Match",
  "if-range",
  "if-unmodified-since",
  "range",
  "transfer-encoding",
  "via"
];

module.exports = function sanitizeHeaders(headers) {
  return Object.keys(headers).reduce((memo, key) => {
    const value = headers[key];
    const normalizedKey = key.toLowerCase();

    if (readOnlyHeaders.includes(normalizedKey)) {
        return memo;
    }

    if (memo[normalizedKey] === undefined) {
      memo[normalizedKey] = []
    }

    const valueArray = Array.isArray(value) ? value : [value]

    valueArray.forEach(valueElement => {
      memo[normalizedKey].push({
        key: key,
        value: valueElement
      });
    });

    return memo;
  }, {});
};
```